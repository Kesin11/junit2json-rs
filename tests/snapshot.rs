use std::fs::File;
use std::io::BufReader;
use insta;

#[test]
fn fixtures_snapshot() {
  insta::glob!("fixtures/*.xml", |path| {
    let file = File::open(path).unwrap();
    let reader = BufReader::new(file);
    let testsuites = junit2json::from_reader(reader).unwrap();

    insta::assert_json_snapshot!(testsuites)
  })
}
