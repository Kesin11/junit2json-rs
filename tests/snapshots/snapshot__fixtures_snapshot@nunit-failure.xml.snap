---
source: tests/snapshot.rs
expression: testsuites
input_file: tests/fixtures/nunit-failure.xml
---
{
  "testsuites": {
    "name": "Tutorial.Tests.dll",
    "time": 0.0105824,
    "tests": 4,
    "failures": 1,
    "testsuite": [
      {
        "name": "Tutorial.Tests.dll",
        "tests": 4,
        "failures": 1,
        "errors": 0,
        "time": 0.0105824,
        "skipped": 0,
        "timestamp": "2019-12-31T 09:33:25Z",
        "hostname": "executor://xunit/VsTestRunner2/netcoreapp",
        "testcase": [
          {
            "name": "Tutorial.Tests.PersonClassTest.PersonClass_ToString",
            "classname": "Tutorial.Tests.PersonClassTest",
            "time": 0.0039775,
            "file": "/Users/kesin/github/csharp-tutorial/tests/Tutorial.Tests/bin/Debug/netcoreapp3.1/Tutorial.Tests.dll"
          },
          {
            "name": "Tutorial.Tests.PersonTest.Person_fname",
            "classname": "Tutorial.Tests.PersonTest",
            "time": 0.0063612,
            "file": "/Users/kesin/github/csharp-tutorial/tests/Tutorial.Tests/bin/Debug/netcoreapp3.1/Tutorial.Tests.dll",
            "failure": {
              "message": "Assert.Equal() Failure\n          ↓ (pos 0)\nExpected: foge\nActual:   john\n          ↑ (pos 0)",
              "type": "failure",
              "inner": "at Tutorial.Tests.PersonTest.Person_fname() in /Users/kesin/github/csharp-tutorial/tests/Tutorial.Tests/PersonTest.cs:line 23"
            }
          },
          {
            "name": "Tutorial.Tests.PersonTest.Person_Name",
            "classname": "Tutorial.Tests.PersonTest",
            "time": 0.0001106,
            "file": "/Users/kesin/github/csharp-tutorial/tests/Tutorial.Tests/bin/Debug/netcoreapp3.1/Tutorial.Tests.dll"
          },
          {
            "name": "Tutorial.Tests.PersonTest.Person_lname",
            "classname": "Tutorial.Tests.PersonTest",
            "time": 0.0001331,
            "file": "/Users/kesin/github/csharp-tutorial/tests/Tutorial.Tests/bin/Debug/netcoreapp3.1/Tutorial.Tests.dll"
          }
        ]
      }
    ]
  }
}
